### PR TITLE
Sc/holidays

### PR DESCRIPTION
**Pull request by sharonco-matrix**
Pull request test to matrix israel
Branch where holidays for handling clinic holidays

### CODE REVIEW

File: interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html [hunk 1]
```diff
@@ -441,7 +454,7 @@
                 if ($e1['eid'] == "") { continue; }
 
                 // skip events for other providers
-                if ($providerid != $e1['aid']) { continue; }
+                if ($providerid != $e1['aid']  && $event['aid']!=0) { continue; }
```
Comment (olegsv) [inline interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html:459]: Describe business logic in comments.

File: interface/main/holidays/Holidays_Storage.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        $res=sqlStatement($sql);
+        while ($row = sqlFetchArray($res)) {
+            $holidays[] = $row;
+        }
+        return $holidays;
+    }
+
+    public function get_holidays_by_dates($star_date,$end_date){
+        $holidays= array();
+        $sql = sprintf('SELECT * FROM %s WHERE (pc_catid="6" OR pc_catid="7") AND pc_eventDate >= "%s" AND pc_eventDate <= "%s"', 'openemr_postcalendar_events',$star_date,$end_date);
```
Comment (olegsv) [inline interface/main/holidays/Holidays_Storage.php]: Don't use hardcoded IDs

File: interface/main/holidays/Holidays_Storage.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                "pc_catid,  pc_aid, pc_pid, pc_title, pc_time, " .
+                "pc_eventDate,  pc_duration, " . "pc_recurrspec,  pc_alldayevent, " . " pc_eventstatus, pc_facility,pc_sharing" .
+                ") VALUES (?,?,?,?,NOW(),?,?,?,?,?,?,?)",
+                $row
+            );
+        }
+
+    }
+
+    public function import_holidays($file){
```
Comment (olegsv) [inline interface/main/holidays/Holidays_Storage.php:91]: Write comments before the function

File: library/calendar.inc [hunk 4]
```diff
@@ -206,4 +206,17 @@ function is_weekend_day($day){
...
+ * @return boolean
+ */
+function is_holiday($date){
+	$holidays =array();
+	require_once($GLOBALS['incdir']."/main/holidays/Holidays_Controller.php");
+	$holidays_controller = new Holidays_Controller();
+	return $holidays_controller->is_holiday($date);
+	
+}
+
```
Comment (olegsv) [inline library/calendar.inc:220]: 1. function comment ?
2. move required_once to top
3. use one instance of Holidays_Controller

File: sql/4_2_2-to-4_3_1_upgrade.sql [hunk 5]
```diff
@@ -181,3 +181,21 @@ CREATE TABLE `valueset` (
...
+
+#IfNotRow openemr_postcalendar_categories pc_catid 6
+INSERT INTO `openemr_postcalendar_categories` VALUES (6, 'Holidays', '#9676DB', 'Clinic holiday', '0', NULL, 'a:5:{s:17:\"event_repeat_freq\";s:1:\"1\";s:22:\"event_repeat_freq_type\";s:1:\"4\";s:19:\"event_repeat_on_num\";s:1:\"1\";s:19:\"event_repeat_on_day\";s:1:\"0\";s:20:\"event_repeat_on_freq\";s:1:\"0\";}', 0, 0, 1, 3, 2, 0, 0, 2);
+#EndIf
+
+#IfNotRow openemr_postcalendar_categories pc_catid 7
+INSERT INTO `openemr_postcalendar_categories` VALUES (7, 'Closed', '#F123A5', 'Clinic closed', '0', NULL, 'a:5:{s:17:\"event_repeat_freq\";s:1:\"1\";s:22:\"event_repeat_freq_type\";s:1:\"4\";s:19:\"event_repeat_on_num\";s:1:\"1\";s:19:\"event_repeat_on_day\";s:1:\"0\";s:20:\"event_repeat_on_freq\";s:1:\"0\";}', 0, 0, 1, 3, 2, 0, 0, 2);
+#EndIf
+
+
```
Comment (olegsv) [inline sql/4_2_2-to-4_3_1_upgrade.sql:630]: remove backslashes

File: interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html (hunk 1 above)
Comment (sharonco-matrix) [inline interface/main/calendar/modules/PostCalendar/pntemplates/default/views/day/ajax_template.html:459]: Done in 3 files (ajax_template)

File: interface/main/holidays/Holidays_Storage.php (hunk 2 above)
Comment (sharonco-matrix) [inline interface/main/holidays/Holidays_Storage.php]: Fixed!

File: interface/main/holidays/Holidays_Storage.php (hunk 3 above)
Comment (sharonco-matrix) [inline interface/main/holidays/Holidays_Storage.php:91]: Done!

File: library/calendar.inc (hunk 4 above)
Comment (sharonco-matrix) [inline library/calendar.inc:220]: Done

File: sql/4_2_2-to-4_3_1_upgrade.sql (hunk 5 above)
Comment (sharonco-matrix) [inline sql/4_2_2-to-4_3_1_upgrade.sql:630]: Done

**Comment by bradymiller** [comment]
Hi,
Just let me know when this is ready for a community code review.
-brady

File: interface/main/holidays/Holidays_Controller.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    public $storage;
+    public $target_file;
+
+    function __construct(){
+        $this->set_target_file();
+        $this->storage = new Holidays_Storage();
+    }
+
+    public function set_target_file(){
+            $this->target_file = $GLOBALS['OE_SITE_DIR']."/". self::UPLOAD_DIR."/".self::FILE_NAME;
```
Comment (olegsv) [inline interface/main/holidays/Holidays_Controller.php:24]: two slashes?

File: library/calendar.inc [hunk 7]
```diff
@@ -206,4 +209,18 @@ function is_weekend_day($day){
...
+/**
+ * This function checks if a certain date (YYYY/MM/DD) is a marked as a holiday/closed  event in the events table
+ * @param (int) $day
+ * @return boolean
+ */
+function is_holiday($date){
+	$holidays =array();
+	if(!$holidays_controller) {
+		$holidays_controller = new Holidays_Controller();
+	}
```
Comment (olegsv) [inline library/calendar.inc]: This function is being called in a loop for each of the days. We don't need to instantiate Holiday_Controller each time.

Comment (sharonco-matrix) [inline library/calendar.inc]: Fixed!

**Pull request closed without merging**